### PR TITLE
FB-772

### PR DESCRIPTION
**Pull request by sequencemedialimited**
[FB-772](https://dsdmoj.atlassian.net/browse/FB-772) -- **Move `schemaUtils` from `fb-components` to `fb-runner-node`**

- Finally, `fb-specification` is completely dead
- This moves the `specification` code from *Components* (where it was put temporarily) into *Runner* where it's actually used

### CODE REVIEW

File: lib/service-data/specification/validate-schema.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+  validateSchema('definition.data', {specs, valid: doubleError, allErrors: false})
+    .then(result => {
+      const expected = getExpected('specs-expected-double.error.show-first')
+      updateFixedPaths(result.valid)
+      t.deepEqual(result, expected, 'it should only report the first error encountered')
+    })
+})
+
+// test('When validating a file that determines its schema automatially', t => {
```
Comment (asmega) [inline lib/service-data/specification/validate-schema.unit.spec.js]: can you delete this or implement the test?

Comment (sequencemedialimited) [reply lib/service-data/specification/validate-schema.unit.spec.js]: Yep! I'll delete (it's one of AR's, so it's probably been commented out for a while)

File: bin/service-data/specification/lint-schemas.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      } catch (e) {
+        reportError([`Invalid json file - ${file}`, e.toString()].join('\n'))
+      }
+      resolve()
+    })
+  })
+}
+
+const testJSONFiles = (files, options) => Promise.all(files.map(file => testJSON(file, options)))
+// , {_id: true}
```
Comment (asmega) [inline bin/service-data/specification/lint-schemas.js]: delete this line?

File: bin/service-data/specification/get-schemas.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  const schemas = []
+
+  try {
+    const {
+      specifications
+    } = require(path.join(specsPath, 'package.json'))
+    specifications.path = specsPath
+    schemas.push(specifications)
+  } catch (e) {
+    // no package.json
```
Comment (asmega) [inline bin/service-data/specification/get-schemas.js]: is it worthwhile logging a warning, or is that not useful?

Comment (sequencemedialimited) [reply bin/service-data/specification/get-schemas.js]: I've wondered about that. Ordinarily, I would always log, but AR doesn't. It's probably an idea to start changing that

File: lib/service-data/specification/validate-schema.unit.spec.js (hunk 1 above)
Comment (sequencemedialimited) [reply lib/service-data/specification/validate-schema.unit.spec.js]: Done

File: bin/service-data/specification/lint-schemas.js (hunk 2 above)
Comment (sequencemedialimited) [reply bin/service-data/specification/lint-schemas.js]: Done

File: bin/service-data/specification/get-schemas.js (hunk 3 above)
Comment (sequencemedialimited) [reply bin/service-data/specification/get-schemas.js]: Done. New rule: delete AR's commented code, add logging in his exceptions

Comment (sequencemedialimited) [reply bin/service-data/specification/get-schemas.js]: (I try not to fiddle with his code too much or else I'll never stop, but I had fiddled in one of those chunks and I still didn't fix the logging, doh)

**Pull request merged**